### PR TITLE
Add initial support for  _resultDependsOnSelf

### PR DESCRIPTION
**Pull request by meg-gupta**
These attributes are used to establish lifetime dependence between argument/self and the result.

Matching swift-syntax PR :  https://github.com/apple/swift-syntax/pull/2281

### CODE REVIEW

**Review by ahoppen — COMMENTED** [review]
This change will need a corresponding PR in swift-syntax to add support for parsing `_resultDependsOn` in the new parser as well.

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

**Comment by hborla** [comment]
Please gate the implementation of a underscored attributes behind an experimental feature flag; see the `EXPERIMENTAL_FEATURE` macro in Features.def for examples.

**Comment by atrick** [comment]
I'll add the NonEscapableTypes experimental feature flag in this PR: https://github.com/apple/swift/pull/69164

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

**Comment by atrick** [comment]
PR https://github.com/apple/swift/pull/69164 landed. You can use the experimental feature flag here.

File: test/Sema/result_depends_on_errors.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// RUN: %target-typecheck-verify-swift  -enable-builtin-module
```
Comment (atrick) [inline test/Sema/result_depends_on_errors.swift]: Hopefully this test now fails without `-enable-experimental-feature NonEscapableTypes`.
Then you'll need `REQUIRES: asserts`

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

**Comment by meg-gupta** [comment]
Previous run of tests passed, but main has changed, so testing again..

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

**Pull request merged**